File: leetcode/238-Product-of-Array-Except-Self.rs
impl Solution {
    pub fn product_except_self(nums: Vec<i32>) -> Vec<i32> {

        let mut answers: Vec<i32> = vec![1; nums.len()];

        let mut left = 1;
        let mut right = 1;

        for i in (0..nums.len()) {
            answers[i] *= left;
            left *= nums[i];
        }
        for i in (0..nums.len()).rev() {
            answers[i] *= right;
            right *= nums[i];
        }
        return answers;
    }
}

File: leetcode/238-Product-of-Array-Except-Self.py
import numpy as np

class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:

        answer = [1 for i in range(len(nums))]
        l, r = 1, 1

        for i in range(len(nums)):
            answer[i] *= l
            l *= nums[i]

        for i in range(len(nums) - 1, -1, -1):
            answer[i] *= r
            r *= nums[i]

        return answer